content/fr/docs/contribute/generate-ref-docs/kubernetes-api.md

```diff
@@ -0,0 +1,337 @@
+
+Allez sur `<k8s-base>` et exécutez ces scripts:
+
+```shell
+hack/update-generated-swagger-docs.sh
+hack/update-swagger-spec.sh
+hack/update-openapi-spec.sh
+hack/update-generated-protobuf.sh
+hack/update-api-reference-docs.sh
+```
+
+Exécutez `git status` pour voir ce qui a été généré.
+
+```shell
+On branch master
+...
+    modified:   api/openapi-spec/swagger.json
+    modified:   api/swagger-spec/apps_v1.json
+    modified:   docs/api-reference/apps/v1/definitions.html
+    modified:   staging/src/k8s.io/api/apps/v1/generated.proto
+    modified:   staging/src/k8s.io/api/apps/v1/types.go
+    modified:   staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go
+```
+
+Voir le contenu de `api/openapi-spec/swagger.json` pour vous assurer que la faute de frappe est corrigée.
+Par exemple, vous pouvez exécuter `git diff -a api/openapi-spec/swagger.json`.
+Ceci est important, car `swagger.json` sera l’entrée de la seconde étape du processus de génération de doc.
+
+Exécutez `git add` et` git commit` pour valider vos modifications.
+Vous avez maintenant deux validations: une avec le fichier `types.go` édité et une avec les spécifications OpenAPI générées et les fichiers associés.
+Gardez ces deux commits séparés.
+C'est-à-dire, ne faites pas un squash de vos commits.
+
+Soumettez vos modifications en tant que [pull request](https://help.github.com/articles/creating-a-pull-request/) à la branche principale du dépôt [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes).
+Surveillez votre pull request, et répondre aux commentaires des relecteurs au besoin.
+Continuez à surveiller votre pull request jusqu'à ce qu'il ait été mergé.
+
+[PR 57758](https://github.com/kubernetes/kubernetes/pull/57758) est un exemple de demande d'extraction qui corrige une faute de frappe dans le code source de Kubernetes.
+
+{{< note >}}
+Il peut être difficile de déterminer le fichier source correct à modifier.
+Dans l'exemple précédent, le fichier source faisant autorité se trouve dans le répertoire `staging` du dépôt `kubernetes/kubernetes`.
+Mais dans votre cas, le répertoire `staging` pourrait ne pas être l’endroit où trouver la source faisant autorité.
+Pour vous guider, consultez les fichiers `README` dans le dépôt [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes/tree/master/staging) et dans des dépôt comme [kubernetes/apiserver](https://github.com/kubernetes/apiserver/blob/master/README.md).
+{{< /note >}}
+
+### Cherry picking votre commit dans une branche release
+
+Dans la section précédente, vous avez modifié un fichier dans la branche principale, puis exécuté des scripts pour générer une spécification OpenAPI et les fichiers associés.
+Vous avez ensuite soumis vos modifications dans une demande d'extraction à la branche maître du dépôt `kubernetes/kubernetes`.
+Supposons maintenant que vous souhaitiez backporter votre modification dans une branche de publication.
```

Comment:
Up to you if we should to change the word "backporter" -> "reporter"

Reply:
J'ai éviter de créer un verbe et j'ai utilisé faire un backport